src/snapping/util.js

```diff
@@ -0,0 +1,88 @@
+const turf = require("@turf/turf");
+
+const LINE_TYPES = ["line", "fill", "fill-extrusion"];
+const CIRCLE_TYPES = ["circle", "symbol"];
+
+const getBufferLayerType = rootLayer => {
+  if (LINE_TYPES.includes(rootLayer.type)) {
+    return "line";
+  } else if (CIRCLE_TYPES.includes(rootLayer.type)) {
+    return "circle";
+  } else {
+    console.error(
+      `Unsupported snap layer type ${rootLayer.type} for layer ${rootLayer.id}`
+    );
+  }
+};
+
+const shouldSnapToVertex = (
+  hoverPoint,
+  nearestPoint,
+  enpoint,
+  vertexPullFactor
+) => {
+  const smallerDistance = turf.distance(hoverPoint, nearestPoint);
+  const largerDistance = turf.distance(hoverPoint, enpoint);
+  return largerDistance / smallerDistance < vertexPullFactor;
+};
+
+exports.getBufferLayerId = layerId => `_snap_buffer_${layerId}`;
+
+exports.getBufferLayer = (bufferLayerId, rootLayer, snapDistance) => {
+  const bufferLayer = {
+    id: bufferLayerId,
+    source: rootLayer.source
+  };
+
+  bufferLayer.type = getBufferLayerType(rootLayer);
+
+  if (rootLayer.sourceLayer) {
+    bufferLayer["source-layer"] = rootLayer.sourceLayer;
+  }
+  if (rootLayer.filter) {
+    bufferLayer.filter = rootLayer.filter;
+  }
+  if (bufferLayer.type === "circle") {
+    bufferLayer.paint = {
+      "circle-color": "hsla(0,100%,50%,0.001)",
```

Comment:
0.001 seems like very transparent... Can that even be seen at all?

Reply:
Yeah I just let that one go  as far as investigating Y its like that. Presumably theres a reason. yeah no thats definetly not actually visible.